statsd/src/main/java/org/kaazing/monitoring/client/MainApp.java

```diff
@@ -21,35 +21,69 @@
 
 package org.kaazing.monitoring.client;
 
+import java.io.IOException;
+import java.util.Locale;
 import java.util.concurrent.atomic.AtomicBoolean;
 
-import org.kaazing.monitoring.client.api.Metric;
-import org.kaazing.monitoring.client.api.MetricsCollector;
-import org.kaazing.monitoring.client.impl.MetricsCollectorAgrona;
+import org.kaazing.monitoring.reader.MetricsCollectorFactory;
+import org.kaazing.monitoring.reader.api.Metric;
+import org.kaazing.monitoring.reader.api.MetricsCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class MainApp {
 
-    private static final String IP = "monitoring";
-    private static final int PORT = 10000;
+    private static final int DEFAULT_UPDATE_INTERVAL = 1000;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MainApp.class);
 
     public static void main(String[] args) {
 
-        StatsdPublisher client;
+        StatsdPublisher client = null;
+        Configuration config = new Configuration();
+
+        if (!config.loadConfigFile()) {
+            LOGGER.error("There was a problem with the configuration file. Exiting application.");
+            System.exit(1);
+        }
+
+        String hostname = config.get(Configuration.CFG_STATSD_HOST);
+        int port = Integer.parseInt(config.get(Configuration.CFG_STATSD_PORT));
+        int updateInterval =
+                Integer.parseInt(config.get(Configuration.CFG_UPDATE_INTERVAL, Integer.toString(DEFAULT_UPDATE_INTERVAL)));
+
+        try {
+            client = new StatsdPublisher(hostname, port);
+        } catch (IOException e) {
+            LOGGER.error(String.format("There was a problem initializing the StatsD publisher. Exiting application.", e));
+            System.exit(1);
+        }
+
+        final AtomicBoolean running = new AtomicBoolean(true);
+
+        MetricsCollector metricsCollector = MetricsCollectorFactory.getInstance();
+
+        if (metricsCollector == null) {
+            LOGGER.error("There was a problem initializing the metrics reader. Exiting application.");
+            System.exit(1);
+        }
 
         try {
-            client = new StatsdPublisher(IP, PORT);
-            final AtomicBoolean running = new AtomicBoolean(true);
+            // Waits until the metrics file is created
+            while (metricsCollector.initialize() == false) {
+                Thread.sleep(updateInterval);
```

Comment:
What is the asynchronous operation in initialize() that you are waiting for?  When it fails the first time, what will change such that it'll succeed in a subsequent attempt?  I'm not quite understanding this loop, any insights you can provide would be appreciated.